app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/DataMapperStepHandler.java

```diff
@@ -35,8 +58,126 @@ public boolean canHandle(Step step) {
     public Optional<ProcessorDefinition<?>> handle(Step step, ProcessorDefinition<?> route, IntegrationRouteBuilder builder, String flowIndex, String stepIndex) {
         ObjectHelper.notNull(route, "route");
 
-        return Optional.of(
-            route.toF("atlas:mapping-flow-%s-step-%s.json?encoding=UTF-8&sourceMapName=" + OutMessageCaptureProcessor.CAPTURED_OUT_MESSAGES_MAP, flowIndex, stepIndex)
-        );
+        List<Map<String, Object>> dataSources = getAtlasmapDataSources(step.getConfiguredProperties());
+
+        route = handleJsonTypeSourceProcessor(route, dataSources).orElse(route);
+        route = route.toF("atlas:mapping-flow-%s-step-%s.json?encoding=UTF-8&sourceMapName=" + OutMessageCaptureProcessor.CAPTURED_OUT_MESSAGES_MAP, flowIndex, stepIndex);
+        route = handleJsonTypeTargetProcessor(route, dataSources).orElse(route);
```

Comment:
I wouldn't use `Optional` to return from private methods makes the logic hard to read IMHO.

Reply:
Totally agree, thanks! Pushed improvements to this PR